src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/CallbackControllerFT.java

```diff
@@ -145,284 +49,258 @@ public class CallbackControllerFT {
-        when(userService.getUserDetails(anyString())).thenReturn(UserDetails.builder().roles(List.of("caseworker-privatelaw-solicitor")).build());
-        when(roleAssignmentApi.updateRoleAssignment(any(), any(), any(), any(RoleAssignmentRequest.class)))
-            .thenReturn(RoleAssignmentResponse.builder().build());
-        mockMvc.perform(post("/send-to-gatekeeper")
-                            .contentType(MediaType.APPLICATION_JSON)
-                            .header("Authorization", idamTokenGenerator.generateIdamTokenForSolicitor())
-                            .header("ServiceAuthorization", serviceAuthenticationGenerator.generateTokenForCcd())
-                            .content(requestBody)
-                            .accept(MediaType.APPLICATION_JSON))
-            .andExpect(status().isOk())
-            .andReturn();
+
+        request
+            .header("Content-Type", APPLICATION_JSON_VALUE)
+            .header("Accepts", APPLICATION_JSON_VALUE)
+            .header("Authorization", idamTokenGenerator.generateIdamTokenForSystem())
+            .header("ServiceAuthorization", serviceAuthenticationGenerator.generateTokenForCcd())
+            .body(requestBody)
+            .when()
+            .contentType(APPLICATION_JSON_VALUE)
+            .post("/send-to-gatekeeper")
+            .then()
+            .assertThat().statusCode(200)
+            .body("data.gatekeepingDetails", notNullValue());
 
     }
 
     @Test
     public void givenC100Case_whenRpaResent_then200Response() throws Exception {
         String requestBody = ResourceLoader.loadJson(C100_RESEND_RPA);
-        when(authorisationService.isAuthorized(any(),any())).thenReturn(true);
-        mockMvc.perform(post("/update-party-details")
-                            .contentType(MediaType.APPLICATION_JSON)
-                            .header("Authorization", idamTokenGenerator.generateIdamTokenForSolicitor())
-                            .header("ServiceAuthorization", serviceAuthenticationGenerator.generateTokenForCcd())
-                            .content(requestBody)
-                            .accept(MediaType.APPLICATION_JSON))
-            .andExpect(status().isOk())
-            .andReturn();
-
+        request
+            .header("Content-Type", APPLICATION_JSON_VALUE)
+            .header("Accepts", APPLICATION_JSON_VALUE)
+            .header("Authorization", idamTokenGenerator.generateIdamTokenForSystem())
+            .header("ServiceAuthorization", serviceAuthenticationGenerator.generateTokenForCcd())
+            .body(requestBody)
+            .when()
+            .contentType(APPLICATION_JSON_VALUE)
+            .post("/update-party-details")
+            .then()
+            .assertThat().statusCode(200)
+            .body("data.caseTypeOfApplication", equalTo("C100"),
+                  "data.caApplicant1Policy.OrgPolicyCaseAssignedRole",  equalTo("[C100APPLICANTSOLICITOR1]"),
```

Comment:
What are we asserting here? and the name of the method has any relation with this?